### PR TITLE
Bug/#4435/tariff page

### PR DESCRIPTION
**Pull request by melnychenkoas**
Change station dropdown to dropdown with a checkboxes; 
Return edit location name request to the previous state;
Remove redundant code

### CODE REVIEW

**Review by NewArtMask — APPROVED** [review]

**Review by voronkooleksandr — APPROVED** [review]

File: src/app/ubs/ubs-admin/components/ubs-admin-tariffs/ubs-admin-tariffs-location-dashboard.component.ts [hunk 1]
```diff
@@ -156,41 +160,57 @@ export class UbsAdminTariffsLocationDashboardComponent implements OnInit, OnDest
...
+      requestAnimationFrame(() => {
+        trigger.openPanel();
+      });
+    }
   }
 
-  isEmpty(): boolean {
-    return this.filteredLocations.length === 0;
+  public checkStation(item): boolean {
+    return this.selectedStation.indexOf(item) >= 0;
```
Comment (hnativlyubomyr) [inline src/app/ubs/ubs-admin/components/ubs-admin-tariffs/ubs-admin-tariffs-location-dashboard.component.ts:188]: use includes here:
this.selectedStation.includes(item)

File: src/app/ubs/ubs-admin/components/ubs-admin-tariffs/ubs-admin-tariffs-location-dashboard.component.ts [hunk 2]
```diff
@@ -156,41 +160,57 @@ export class UbsAdminTariffsLocationDashboardComponent implements OnInit, OnDest
...
   }
 
-  positionsFilter(): void {
-    if (this.checkedCities.length !== 0) {
-      this.filteredLocations = this.onPositionSelected();
+  public setStationPlaceholder(): void {
+    if (this.selectedStation.length) {
+      this.stationPlaceholder = this.selectedStation.length + ' вибрано';
```
Comment (hnativlyubomyr) [inline src/app/ubs/ubs-admin/components/ubs-admin-tariffs/ubs-admin-tariffs-location-dashboard.component.ts:193]: use interpolation string:
`${this.selectedStation.length} вибрано`;

File: src/app/ubs/ubs-admin/components/ubs-admin-tariffs/ubs-admin-tariffs-location-dashboard.component.ts [hunk 3]
```diff
@@ -200,6 +220,17 @@ export class UbsAdminTariffsLocationDashboardComponent implements OnInit, OnDest
     }
   }
 
+  toggleSelectAllStation() {
```
Comment (hnativlyubomyr) [inline src/app/ubs/ubs-admin/components/ubs-admin-tariffs/ubs-admin-tariffs-location-dashboard.component.ts:223]: add return type - void

**Review by hnativlyubomyr — APPROVED** [review]

**Review by Myh45 — APPROVED** [review]

File: src/app/ubs/ubs-admin/components/ubs-admin-tariffs/ubs-admin-tariffs-location-dashboard.component.ts [hunk 4]
```diff
@@ -156,41 +160,57 @@ export class UbsAdminTariffsLocationDashboardComponent implements OnInit, OnDest
     }
   }
 
-  isChecked(): boolean {
-    return this.checkedCities.length === this.cities.length;
+  public onSelectStation(event: MatAutocompleteSelectedEvent): void {
+    const newValue = event.option.value;
+    if (this.selectedStation.includes(newValue)) {
+      this.selectedStation = [...this.selectedStation.filter((item) => item !== newValue)];
```
Comment (Mukola1983) [inline src/app/ubs/ubs-admin/components/ubs-admin-tariffs/ubs-admin-tariffs-location-dashboard.component.ts:166]: it seems spread operator here not needed)

this.selectedStation = this.selectedStation.filter((item) => item !== newValue)  - it should work

**Review by Mukola1983 — APPROVED** [review]

**Pull request merged**